### DISCUSSION TITLE
license.expired webhook event

### DISCUSSION THREAD

**Discussion by skar-sko** [keygen-sh/community#131] [Q&A] [answered]
Hi,
Are there some hidden rules for this event? Like can it only fire once per license, even if it is renewed after the event has occurred? I'm trying to get it firing again using a somewhat brute force approach by updating the license expiry date through postman. I seems that it fires on new licenses but not when setting a new expiration date after the fact and letting it expire again.
Thanks for a great product by the way!

**Comment by ezekg** [comment, marked as answer]
It will not be fired within 24 hours of the last `license.expired` webhook event, because a policy cannot have a duration shorter than 1 day. Forcing the event via updating the license's expiry attribute out of band isn't something we currently support. You can read the actual code [here](https://github.com/keygen-sh/keygen-api/blob/master/app/workers/license_expirations_worker.rb). What's your use case for forcing the event?

**Reply by skar-sko** [reply]
Right, makes perfect sense. I don't have an actual use case for it. It's only something I found when running tests for a flow that is supposed to be triggered by this event while using a test key. For simplicity I was using the same key and thought to update it just to have it trigger again, just for the sake of testing. Thanks for clarifying and the link to the code!
